src/Raven.Server/Documents/AttachmentsStorage.cs

```diff
@@ -997,5 +1030,25 @@ public static void AssertAttachments(BlittableJsonReaderObject document, Documen
             }
         }
 #endif
+        public void AssertAttachmentsFromReplication(DocumentsOperationContext context, string id, BlittableJsonReaderObject document)
+        {
+            if (document.TryGet(Constants.Documents.Metadata.Key, out BlittableJsonReaderObject metadata) &&
+                metadata.TryGet(Constants.Documents.Metadata.Attachments, out BlittableJsonReaderArray attachments))
+            {
```

Comment:
Let's get everything about the document itself out of here, we shouldn't have the document's metadata details in attachment storage